DigitalLearningSolutions.Data.Tests/Services/TutorialContentServiceTests.cs

```diff
@@ -46,10 +46,88 @@ public void Get_tutorial_information_should_return_tutorial()
                 "<ul><li>use the Go To feature to jump to a particular page</li><li>browse a document by a specific element</li></ul></body></html>",
                 "/MOST/Word07Core/swf/1_1_02_Navigate_documents.swf",
                 "/MOST/Word07Core/MOST_Word07_1_1_02.dcr",
-                "/MOST/Word07Core/support.html?popup=1&item=navigateDocs"
+                "/MOST/Word07Core/support.html?popup=1&item=navigateDocs",
+                "https://www.dls.nhs.uk/tracking/MOST/Word07Core/Assess/L2_Word_2007_Post_1.dcr",
+                51,
+                75
             ));
         }
 
+        [Test]
+        public void Get_tutorial_information_should_return_null_nextTutorial_if_last_tutorial_in_section()
+        {
+            // Given
+            const int candidateId = 1;
+            const int customisationId = 1379;
+            const int sectionId = 74;
+            const int tutorialId = 52;
+
+            // When
+            var tutorial = tutorialContentService.GetTutorialInformation(candidateId, customisationId, sectionId, tutorialId);
+
+            // Then
+            tutorial.Should().NotBeNull();
+            tutorial!.NextTutorialId.Should().BeNull();
+        }
+
+        [Test]
+        public void Get_tutorial_information_should_return_null_nextSection_if_last_section_in_course()
+        {
+            // Given
+            const int candidateId = 1;
+            const int customisationId = 1379;
+            const int sectionId = 82;
+            const int tutorialId = 94;
+
+            // When
+            var tutorial = tutorialContentService.GetTutorialInformation(candidateId, customisationId, sectionId, tutorialId);
+
+            // Then
+            tutorial.Should().NotBeNull();
+            tutorial!.NextSectionId.Should().BeNull();
+        }
+
+        [Test]
+        public void Get_tutorial_information_nextTutorial_should_skip_tutorials_not_in_customisation()
+        {
+            // Given
+            const int candidateId = 210934;
+            const int customisationId = 18366;
+            const int sectionId = 973;
+            const int tutorialId = 4257;
+
+            // The next tutorial ID in this section is 4258, but the next tutorial selected in CustomisationTutorials is 4263

```

Comment:
Very useful comment, thanks! I think these tests where we're relying on the state of the db is a case where comments can be really useful 🙂

Reply:
Yeah absolutely, and it was a bit ridiculous when I tried to encode all of this in the function title! 🙂